pkg/gadgets/trace/oomkill/tracer/tracer_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2022-2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+// +build linux
+
+package tracer_test
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/sys/unix"
+
+	utilstest "github.com/inspektor-gadget/inspektor-gadget/internal/test"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/oomkill/tracer"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/oomkill/types"
+)
+
+func TestOomkillTracerCreate(t *testing.T) {
+	t.Parallel()
+
+	utilstest.RequireRoot(t)
+
+	tracer := createTracer(t, &tracer.Config{}, func(*types.Event) {})
+	require.NotNil(t, tracer, "Returned tracer was nil")
+}
+
+func TestOomkillTracerStopIdempotent(t *testing.T) {
+	t.Parallel()
+
+	utilstest.RequireRoot(t)
+
+	tracer := createTracer(t, &tracer.Config{}, func(*types.Event) {})
+
+	// Check that a double stop doesn't cause issues
+	tracer.Stop()
+	tracer.Stop()
+}
+
+func TestOomkillTracer(t *testing.T) {
+	t.Parallel()
+
+	utilstest.RequireRoot(t)
+
+	_, err := os.Stat("/sys/fs/cgroup/cgroup.controllers")
+	if err != nil {
+		t.Skip("This test requires cgroup v2 to be mounted: %w", err)
+	}
+
+	cgroupPath := "/sys/fs/cgroup/test-oomkill-tracer"
+	memoryLimit := 512 * 1024 // 512 KB.
+
+	// The whole idea of this test is to spawn a child which will allocate so much
+	// memory that it get killed by the OOM killer, hence generating an event.
+	// We cannot do this in the root cgroup, as it could lead to misbehavior from
+	// other test as they would not be able to allocate memory.
+	// So, the whole idea is to create a memory cgroup, set the memory.max limit,
+	// spawn a process, subscribe this child to the memory cgroup and then
+	// allocate the memory until it get killed.
+	err = os.Mkdir(cgroupPath, 0o755)
+	require.NoError(t, err, "creating cgroup path %s", cgroupPath)
+	defer syscall.Rmdir(cgroupPath)
```

Comment:
I would also run a `syscall.Rmdir` at the beginning of the test to clean up previous executions of the test that were interrupted without running the defer.